components/script/dom/gpudevice.rs

```diff
@@ -708,6 +849,49 @@ impl GPUDeviceMethods for GPUDevice {
 
         GPURenderPipeline::new(&self.global(), render_pipeline, self.device, valid)
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-pusherrorscope
+    fn PushErrorScope(&self, filter: GPUErrorFilter) {
+        let scope_id = self.next_scope_id.get();
+        self.next_scope_id.set(scope_id + 1);
+        let err_scope = ErrorScopeInfo {
+            filter,
+            op_count: 0,
+            error: None,
+            promise: None,
+        };
+        let res = self.error_scopes.borrow_mut().insert(scope_id, err_scope);
+        self.scope_stack.borrow_mut().push(scope_id);
+        assert!(res.is_none());
+    }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-poperrorscope
+    fn PopErrorScope(&self, comp: InRealm) -> Rc<Promise> {
+        let promise = Promise::new_in_current_realm(&self.global(), comp);
+        let scope_id = if let Some(e) = self.scope_stack.borrow_mut().pop() {
+            e
+        } else {
+            promise.reject_error(Error::Operation);
+            return promise;
+        };
+        let mut err_scope;
+        {
+            err_scope = self.error_scopes.borrow_mut().remove(&scope_id).unwrap();
```

Comment:
if we remove it here on the client side, what are we going to do with an error coming from the server with this "scope_id"?

Reply:
We reinsert it(line 890) in case we have more results to be captured